src/interfaces/mining.h

```diff
@@ -46,6 +72,9 @@ class Mining
      */
     virtual std::unique_ptr<node::CBlockTemplate> createNewBlock(const CScript& script_pub_key, bool use_mempool = true) = 0;
 
+    // Implemented in https://github.com/bitcoin/bitcoin/pull/30356
+    virtual std::unique_ptr<BlockTemplate> createNewBlock2(const CScript& script_pub_key, const node::BlockCreateOptions& options={}) { return {}; }
```

Comment:
Is it possible to rename this to `createNewBlock` and drop the one above? Or is there an inconsistency between my PRs?

Reply:
re: https://github.com/bitcoin/bitcoin/pull/30437#discussion_r1681055963

> Is it possible to rename this to `createNewBlock` and drop the one above? Or is there an inconsistency between my PRs?

It's not really possible without making the PR bigger because mining code relies on createNewBlock returning CBlockTemplate not BlockTemplate, and I want mining code to work and CI to pass. 

The purpose of createNewBlock2 is just to make sure serialization works when createNewBlock is changed to return BlockTemplate.